internal/dockerutil/image.go

```diff
@@ -87,14 +87,27 @@ type ContainerOptions struct {
 	User string
 }
 
+// ContainerExecResult is a wrapper type that wraps an exit code and associated output from stderr & stdout, along with
+// an error in the case of some error occurring during container execution.
+type ContainerExecResult struct {
+	Err            error // Err is nil, unless some error occurs during the container lifecycle.
+	ExitCode       int
+	Stdout, Stderr []byte
+}
+
 // Run creates and runs a container invoking "cmd". The container resources are removed after exit.
 //
 // Run blocks until the command completes. Thus, Run is not suitable for daemons or servers. Use Start instead.
 // A non-zero status code returns an error.
-func (image *Image) Run(ctx context.Context, cmd []string, opts ContainerOptions) (stdout, stderr []byte, err error) {
+func (image *Image) Run(ctx context.Context, cmd []string, opts ContainerOptions) ContainerExecResult {
 	c, err := image.Start(ctx, cmd, opts)
 	if err != nil {
-		return nil, nil, err
+		return ContainerExecResult{
+			Err:      err,
+			ExitCode: 1,
```

Comment:
I think typically `-1` would indicate an exceptional case where the execution did not happen.